### DISCUSSION TITLE
Unable to connect to MariaDB from Firebase Function

### DISCUSSION THREAD

**Discussion by abakucz** [prisma/prisma#15475] [Q&A]
### Bug description

Hi, I am not really sure yet, if this is a Prisma bug, but I can't seem to connect to remote MariaDB from Firebase Function on Blaze plan (basically Google Cloud Function).

My connection string is simple: `mysql://schoolsstats:[password]@db.al21.savana-hosting.cz:10076/schoolsstats` 

But every connection attempt ends up with `errorCode: P1001` - `Can't reach database server at db.al21.savana-hosting.cz:10076`.

If i set `pool_timeout=30&connect_timeout=30` , it ends up with `errorCode: P2024` - `Timed out fetching a new connection from the connection pool.`

Now the strange part:
✓ **Local machine** - Connection works perfectly with that same connection string on Firebase Emulator -> **connection string is valid**
✓ **Remote Firebase Function** - I can successfully connect to that DB via [noda-mysql2](https://github.com/sidorares/node-mysql2) package with same credentials -> **nothing is preventing connection from the Firebase Function**

So the question is, why can't I connect from Firebase Function with Prisma while I can with node-mysql2?

I have no clues how to track this one down.

### How to reproduce

Client initialization is simple - `db/client.ts`:
```ts
export const prisma = new PrismaClient();
```

Then a simple query:
```ts
const data = await prisma.school.findMany({ take: 10 });
```

I've even tried to first call:
```ts
await prisma.$connect()
```
but this ends up with the same results.

### Expected behavior

I am able to connect to the remote DB.

### Prisma information

Connection string: `mysql://schoolsstats:[password]@db.al21.savana-hosting.cz:10076/schoolsstats`

Schema
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("STATS_DB_URL")
}

model School {
  id          String        @id @db.VarChar(50)
  name        String        @db.VarChar(255)
  npeTotal    Int           @default(0) @map("npe_total")
  postalCode  String?       @map("postal_code") @db.VarChar(10)
  country     String?       @db.VarChar(10)

  @@index([id], map: "id")
  @@map("schools")
}
```

### Environment & setup

- **OS**: Firebase Function runs on Ubuntu 18.04, my local machine runs on Windows 10
- **Database**: MariaDB
- **Node.js version**: Firebase Function runs on Node 14 (I've even tried Node 16)

### Prisma Version

Firebase Function (by calling `Prisma.prismaVersion`):
```
 client: '4.2.1', 
 engine: '2920a97877e12e055c1333079b8d19cee7f33826'
```

Local machine:
```
Environment variables loaded from .env
prisma                  : 4.2.1
@prisma/client          : 4.2.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 2920a97877e12e055c1333079b8d19cee7f33826
Studio                  : 0.469.0
```

**Comment by danstarns** [comment]
If it works on your local machine then it seems like a connection issue with Firebase, have you tried looking into Firebase "firewall" settings?  Maybe you need to whitelist your IP address to connect.

**Comment by abakucz** [comment]
> If it works on your local machine then it seems like a connection issue with Firebase, have you tried looking into Firebase "firewall" settings? Maybe you need to whitelist your IP address to connect.

To my knowledge, there are no Firebase restrictions on outgoing connections when you're on Blaze plan. Also DB for now accepts incoming connection from any IP.

And most importantly as I noted, I can successfully connect to that DB from that same Firebase Function via [noda-mysql2](https://github.com/sidorares/node-mysql2) package with same credentials. So, **nothing is (most likely) preventing connection from the Firebase Function**.

**Reply by pantharshit00** [reply]
Hello @abakucz 

Sorry for late followup here. 

Can you please share part of your codebase in a Git repo so that we may try to reproduce this?

**Reply by abakucz** [reply]
Hi @pantharshit00, unfortunately I had to eventually switch to CockroachDB, which works just fine with Prisma, so there is no old codebase available at this moment. And since it didn't work I never even committed it to Git repo.

But it was all very basic boilerplate code working just fine on my machine, but not working in Cloud Function.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
